### PR TITLE
When reading from a file, interpret imports relative to that file

### LINKED ISSUE #114: Wrong CWD handling
Consider running `dhall-to-cabal foo/bar.dhall`. The input file name could have relative imports, which have to be resolved relative to `foo/`. However, we read the file and then process it totally divorced of that directory information: we'll try to resolve imports relative to the current working directory.

This isn't trivially fixable in dhall-to-cabal: we need some extra API from dhall itself to specify what directory an expression's imports should be resolved relative to.

### PR DESCRIPTION
**Pull request by quasicomputational**
Closes #114 

This only works against `dhall-haskell` HEAD, so it shouldn't be merged yet. Pre-requisite for #113.

### CODE REVIEW

File: golden-tests/GoldenTests.hs [hunk 1]
```diff
@@ -25,6 +25,11 @@ import DhallLocation ( DhallLocation ( DhallLocation ) )
 import DhallToCabal ( dhallToCabal )
 
 
+
+-- Note: all of the dhall-to-cabal tests use the CWD as their root
+-- directory for imports, not the directory that they live in.
```
Comment (ocharles) [inline golden-tests/GoldenTests.hs]: I actually think this is confusing, and if we're going to fix up cwd stuff, let's do it properly and have all those files have correct references.

**Comment by quasicomputational** [comment]
OK, the tests now do the Right Thing, and this now works with dhall-haskell HEAD. Our API is still changing, but at least we're not juggling multiple `FilePath` arguments any more.

We already transitively depend on lens so adding a direct dependency shouldn't be a problem, even though it's just for `set`.

**Comment by quasicomputational** [comment]
Dhall 1.16 is out and it works locally for me. I think this just needs for the Nix stuff to be bumped.

**Comment by ocharles** [comment]
Sweet, will try and get on top of this in the week.

File: dhall-to-cabal.cabal [hunk 2]
```diff
@@ -222,9 +224,10 @@ test-suite golden-tests
         Cabal ^>=2.2,
         Diff ^>=0.3.4,
         bytestring ^>=0.10,
-        dhall ^>=1.15.0,
+        dhall ^>=1.16.0,
         dhall-to-cabal -any,
         filepath ^>=1.4,
+        lens ^>=4.17,
```
Comment (ocharles) [inline dhall-to-cabal.cabal]: Could we go with microlens? All of `lens` to only use `set` is a bit heavy.

**Comment by ocharles** [comment]
I'm on board with this other than the `microlens` comment.

**Comment by quasicomputational** [comment]
OK, pushed, plus fixing `stack.yaml` that I broke in the process.

Sort of funny that we've got `lens`, `lens-family-core` and `microlens` all in our dependency graph now.

**Pull request merged**